core/src/main/scala/codesearch/core/index/regex/lexer/Tokenizer.scala

```diff
@@ -0,0 +1,40 @@
+package codesearch.core.index.regex.lexer
+
+import fastparse._
+import NoWhitespace._
+import codesearch.core.index.regex.lexer.tokens._
+
+trait TokenizerMethods {
+  protected def leftForOtherSymbols[_: P] = P(CharIn("(", "[", "[^")).!
+
+  protected def rightForOtherSymbols[_: P] = P(CharIn(")", "]")).!
+
+  protected def specialSymbols[_: P] = P(CharIn(" ", "$", "%", "^", "&", "*", "+", "?", "!")).!
```

Comment:
Please make sure all possible special characters are listed here. For example `\` missed here. It's important for us to know if one of this symbols was escaped by `\`.